decidim-comments/app/controllers/decidim/comments/comments_controller.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Comments
+    # Controller that manages the comments for a commentable object.
+    #
+    class CommentsController < Decidim::Comments::ApplicationController
+      before_action :authenticate_user!, only: [:create]
+      before_action :set_commentable
+      before_action :ensure_commentable!
+
+      helper_method :root_depth, :commentable, :order, :reply?, :reload?
+
+      def index
+        @comments = SortedComments.for(
+          commentable,
+          order_by: order,
+          after: params.fetch(:after, 0).to_i
+        )
+
+        render :reload if reload?
+      end
+
+      def create
+        form = Decidim::Comments::CommentForm.from_params(
```

Comment:
To avoid hand crafted POSTs, shouldn't we check if comments are enabled and the user is himself?
I see there's `authenticate_user!` to check if she is logged in, so this is already checked.

Reply:
I have added permission checks that ensure the user is allowed to comment.

There's no reason to check `user == user` because we should trust the `authenticate_user!` as you mentioned.

Reply:
You are right, the author is resolved from the session via `current_user`